src/http_client_async.zig

```diff
@@ -552,6 +557,18 @@ pub const HTTPThread = struct {
     }
 
     fn drainEvents(this: *@This()) void {
+        while (this.queued_shutdowns.pop()) |http| {
+            if (socket_async_http_abort_tracker.get(http.async_http_id)) |socket_ptr| {
```

Comment:
one of the functions lets you get and delete in the same call

I think its

fetchSwapRemove